Assets/_BForBoss/_Character/Scripts/FirstPersonPlayer.cs

```diff
@@ -246,16 +264,41 @@ private void ResetJumpCount()
 
         private void ToggleThirdPerson()
         {
-            cmCrouchedCamera.SetActive(!IsThirdPerson && IsCrouching());
-            cmWalkingCamera.SetActive(!IsThirdPerson && !IsCrouching());
-            cmThirdPersonCamera.SetActive(IsThirdPerson);
+            if (switchingViews != null)
+            {
+                StopCoroutine(switchingViews);
+                switchingViews = null;
+            }
+            cmCrouchedCamera.gameObject.SetActive(!IsThirdPerson && IsCrouching());
+            cmWalkingCamera.gameObject.SetActive(!IsThirdPerson && !IsCrouching());
+            cmThirdPersonCamera.gameObject.SetActive(IsThirdPerson);
+            if (IsThirdPerson)
+            {
+                TogglePlayerModel();
+            }
+            else
+            {
+                switchingViews = StartCoroutine(TogglePlayerModelWithDelay());
+            }
         }
 
         private void SwitchView(InputAction.CallbackContext context)
         {
             if (context.started) IsThirdPerson = !IsThirdPerson;
         }
 
+        private IEnumerator TogglePlayerModelWithDelay()
+        {
+            yield return new WaitForSeconds(SwitchViewDuration);
```

Comment:
Wondering, why have the need to have a coroutine when toggling? Is it to prevent spamming the button?

Reply:
Its so that when you blend from third person to first, you still see the body up until the moment you've completed the blend. In the case of going from first person to third I'm not using the coroutine, I'm just immediately switching

Reply:
This is also why I changed the animator to use a different variable instead of directly relying on the `IsThirdPerson` variable, so that it animates up until the blend is complete